packages/core/src/WorkflowExecute.ts

```diff
@@ -1066,6 +1066,21 @@ export class WorkflowExecute {
 
 								nodeSuccessData = runNodeData.data;
 
+								const didContinueOnFail = nodeSuccessData?.at(0)?.at(0)?.json.error !== undefined;
+
+								while (didContinueOnFail && tryIndex !== maxTries - 1) {
```

Comment:
We're ignoring the `Wait between tries` property with this approach I'm afraid.

Reply:
Please see [fc89956](https://github.com/n8n-io/n8n/pull/9395/commits/fc899562ef28e266b5a55b15a299410341132947)

Reply:
Looks good.

Minor note (no need to refactor) is that this could be a good place to use a `do { ... }. while ()`